modules/markdown/assets/js/jetpack-markdown-block.jsx

```diff
@@ -0,0 +1,70 @@
+/* global React:true */
+
+/**
+ * Internal dependencies
+ */
+import JetpackMarkdownBlockEditor from './jetpack-markdown-block-editor';
+import JetpackMarkdownBlockSave from './jetpack-markdown-block-save';
+
+const { __ } = window.wp.i18n;
+
+const {
+	registerBlockType,
+} = window.wp.blocks;
+
+const el = window.wp.element.createElement;
+
+registerBlockType( 'jetpack/markdown-block', {
+
+	title: __( 'Markdown' ),
+
+	description: [
+		__( 'Write your content in plain-text Markdown syntax.' ),
+		(
+		<p>
+			<a href="https://en.support.wordpress.com/markdown-quick-reference/">
+			Support Reference
+			</a>
+		</p>
+		)
+	],
+
+	icon: el(
+		'svg',
+		{
+			xmlns: 'http://www.w3.org/2000/svg',
+			'class': 'dashicons',
+			width: '208',
+			height: '128',
+			viewBox: '0 0 208 128',
+			stroke: 'currentColor'
+		},
+		el(
+			'rect',
+			{
+				width: '198',
+				height: '118',
+				x: '5',
+				y: '5',
+				ry: '10',
+				'stroke-width': '10',
+				fill: 'none'
+			}
+		),
+		el(
+			'path', { d: 'M30 98v-68h20l20 25 20-25h20v68h-20v-39l-20 25-20-25v39zM155 98l-30-33h20v-35h20v35h20z' }
+		)
+	),
+
+	category: 'formatting',
+
+	attributes: {
+		//The Markdown source is saved in the block content comments delimiter
```

Comment:
What was your reasoning around saving the source in the attributes? Why not in the block's body? Why not in block meta?

Reply:
I thought it was the cleanest solution. It makes super simple to link the markdown source with the block's content (in case you add more than one markdown blocks).
When making blocks I guess we should strive to save the attributes in the block's content, that way we avoid data duplication, but I couldn't find a way to store the markdown source in the block's content and then hide it in the published post (but I'm a total WordPress newbie, maybe there's a way)...
> Why not in block meta?
I guess you meant in the post's meta, right? I guess one of the uses of the post's meta is to share post's info in another parts of WordPress: other plugins, themes, etc. I didn't think the markdown source would be needed in any other place. It would make it harder to manage each markdown block source: if we store the attribute as it is, it'll return a strings array. If we choose to store as json, it'd have to serialize/parse it every time we needed. It seemed way more complex to do it this way.

Reply:
At a glance, I agree with keeping the Markdown source as a regular (HTML-comment-borne) attribute. Aside from ease on the development side, it also gets us data deduplication for free (that is, when rendering the front-end, assuming Gutenberg is stripping HTML comments, the visitor only gets the generated HTML source).

For completion:

- Keeping the Markdown source in the block's body could be achieved using old techniques such as using custom script tags like `<script type="text/markdown">` that the browser can ignore. Gutenberg can be told how to source attributes from these. The downside is that this information is wastefully sent to the visitor.

- It's possible to intercept the block's body to decide what gets sent with the rendered page, but that requires making the block dynamic (and having a PHP render callback to render the block on every page load). I don't see how this option could be better than defaulting to HTML comments for the Markdown source, though.

Reply:
Thanks for the clarification @mcsf ! Especially the last point :+1: